packages/gatsby/src/utils/page-data.ts

```diff
@@ -204,11 +204,15 @@ export async function flush(): Promise<void> {
   }
 
   if (!flushQueue.idle()) {
-    await new Promise(resolve => {
-      flushQueue.drain = resolve as () => unknown
+    await new Promise<void>(resolve => {
+      flushQueue.drain = resolve
     })
   }
 
+  await new Promise<void>(resolve => {
```

Comment:
Confused about the relationship between this & memory usage. Also how the queue would empty almost immediately in your testing? The work of the queue is to write out files which is pretty fast but still should take a number of seconds for 10s of thousands of paths it sounds like your testing.